jupyterlab_gitsync/src/service/notebook_file.ts

```diff
@@ -41,55 +50,100 @@ export class NotebookFile implements IFile {
     return this._dirtyState;
   }
 
+  get activeCell() {
+    return this.content.activeCell;
+  }
+
+  get activeCellIndex(): number {
+    return this.content.activeCellIndex;
+  }
+
   async save() {
     try{
-      await this.context.save();
+      const content = (this.content.model as NotebookModel).toJSON();
+      await this._saveFile(content);
+      this.resolver.addVersion(content, 'base');
     } catch (error) {
       console.warn(error);
```

Comment:
find a way to notify user

Reply:
I'm thinking of displaying the errors somewhere on the UI (maybe clicking the status indicator will show a panel with a log of all service operations, ie. save/sync/etc.), but this will most likely be included in a separate PR.